frontend/src/App.js

```diff
@@ -1,32 +1,43 @@
+import React from "react";
 import {BrowserRouter, Switch, Route} from "react-router-dom";
 import LocationsPage from "./pages/LocationsPage";
 import Header from "./components/Header";
 import {ThemeProvider} from '@material-ui/core/styles';
 import {createMuiTheme} from '@material-ui/core/styles';
 import LocationDetailsPage from "./pages/LocationDetailsPage";
+import {geolocated} from "react-geolocated";
 
-function App() {
-    const darkTheme = createMuiTheme({
-        palette: {
-            mode: 'dark',
-        },
-    });
+class App extends React.Component{
 
-    return (
-        <BrowserRouter>
-            <ThemeProvider theme={darkTheme}>
-                <Header/>
-                <Switch>
-                    <Route path={"/locations"} exact>
-                        <LocationsPage/>
-                    </Route>
-                    <Route path={"/locations/:id"} exact>
-                        <LocationDetailsPage/>
-                    </Route>
-                </Switch>
-            </ThemeProvider>
-        </BrowserRouter>
-    );
+    render() {
+        const darkTheme = createMuiTheme({
```

Comment:
This can be moved outside of the render function, no need to redefine it on every update.